Unable to build kubernetes running 'make'

After merging https://github.com/kubernetes/kubernetes/pull/25978, I am unable to build kubernetes:

```sh
./hack/build-go.sh
NOTE: ./hack/build-go.sh has been replaced by 'make' or 'make all'

The equivalent of this invocation is: 
    make 


make: *** No rule to make target `/*.go', needed by `_output/bin/deepcopy-gen'.  Stop.
```

Before ``eecbfb1a2808b233a4c7032ce26e260276bffde5``:
```sh
./hack/build-go.sh
Go version: go version go1.6.2 linux/amd64
+++ [0713 07:36:27] Building the toolchain targets:
    k8s.io/kubernetes/hack/cmd/teststale
+++ [0713 07:36:27] Building go targets for linux/amd64:
    cmd/kube-dns
...
```

Am I something missing?

```
kubernetes$ ./build/run.sh make
+++ [0713 14:27:38] Verifying Prerequisites....
+++ [0713 14:27:44] Building Docker image kube-build:build-d95f553a94.
+++ [0713 14:27:49] Running build command....
make: *** No targets specified and no makefile found.  Stop.
!!! Error in ./build/../build/common.sh:620
  '"${docker_cmd[@]}" "$@"' exited with status 2
Call stack:
  1: ./build/../build/common.sh:620 kube::build::run_build_command(...)
  2: ./build/run.sh:30 main(...)
Exiting with status 1
```

```
build$ ./run.sh make
./../build/common.sh: Zeile 32: hack/lib/init.sh: Datei oder Verzeichnis nicht gefunden
```


git branch on v1.3.0

@thockin any idea why `make clean all` fails: 
```
$ make clean all
+++ [0714 11:52:49] Verifying Prerequisites....
!!! [0714 11:52:49] Build image not built.  Cannot clean via docker build image.
+++ [0714 11:52:49] Removing data container kube-build-data-0bc7d68a56
+++ [0714 11:52:49] Removing _output directory
+++ [0714 11:52:49] Deleting docker image kube-build:build-0bc7d68a56
+++ [0714 11:52:49] Cleaning all other untagged docker images
Go version: go version go1.7rc1 linux/amd64
+++ [0714 11:52:49] Building the toolchain targets:
    k8s.io/kubernetes/hack/cmd/teststale
+++ [0714 11:52:49] Building go targets for linux/amd64:
    cmd/libs/go2idl/deepcopy-gen
+++ [0714 11:52:51] Placing binaries
make: *** No rule to make target '.make/federation/apis/federation/gofiles.mk.stamp', needed by 'federation/apis/federation/zz_generated.deepcopy.go'.  Stop.
```

Calling `make clean` and `make all` works ok.

--- a/Makefile.generated_files
+++ b/Makefile.generated_files
@@ -198,11 +198,11 @@ DEEPCOPY_FILES := $(addsuffix /$(DEEPCOPY_FILENAME), $(DEEPCOPY_DIRS))
 # in a single run of the tool.
 .PHONY: gen_deepcopy
 gen_deepcopy: $(DEEPCOPY_FILES)
-	if [[ -f $(META_DIR)/$(DEEPCOPY_GEN).todo ]]; then                \
-	    $(DEEPCOPY_GEN)                                               \
+	if [[ -f $(META_DIR)/$(DEEPCOPY_GEN).todo ]]; then                  \
+	    ./hack/run-in-gopath.sh $(DEEPCOPY_GEN)                         \
 	        -i $$(cat $(META_DIR)/$(DEEPCOPY_GEN).todo | paste -sd, -)  \
-	        --bounding-dirs $(PRJ_SRC_PATH)                           \
-	        -O $(DEEPCOPY_BASENAME);                                  \
+	        --bounding-dirs $(PRJ_SRC_PATH)                             \
+	        -O $(DEEPCOPY_BASENAME);                                    \
 	fi
 
 # For each dir in DEEPCOPY_DIRS, this establishes a dependency between the
@@ -243,6 +243,7 @@ $(META_DIR)/$(DEEPCOPY_GEN).mk:
 	     -f '{{range .Deps}}{{.}}{{"\n"}}{{end}}'          \
 	     ./cmd/libs/go2idl/deepcopy-gen                    \
 	     | grep "^$(PRJ_SRC_PATH)"                         \
+	     | sed 's|^$(PRJ_SRC_PATH)|./|'                    \
 	     | xargs go list -e -f '{{.Dir}} {{.Dir}}/*.go');  \
 	echo $$DIRECT $$INDIRECT | sed 's/ / =,/g' | tr '=,' '\n\t'; \
 	) | sed "s|$$(pwd -P)/||" > $@.tmp;                    \
@@ -308,10 +309,10 @@ CONVERSION_FILES := $(addsuffix /$(CONVERSION_FILENAME), $(CONVERSION_DIRS))
 # in a single run of the tool.
 .PHONY: gen_conversion
 gen_conversion: $(CONVERSION_FILES)
-	if [[ -f $(META_DIR)/$(CONVERSION_GEN).todo ]]; then                \
-	    $(CONVERSION_GEN)                                               \
+	if [[ -f $(META_DIR)/$(CONVERSION_GEN).todo ]]; then                  \
+	    ./hack/run-in-gopath.sh $(CONVERSION_GEN)                         \
 	        -i $$(cat $(META_DIR)/$(CONVERSION_GEN).todo | paste -sd, -)  \
-	        -O $(CONVERSION_BASENAME);                                  \
+	        -O $(CONVERSION_BASENAME);                                    \
 	fi
 
 # Establish a dependency between the deps file and the dir.  Whenever a dir
@@ -419,6 +420,7 @@ $(META_DIR)/$(CONVERSION_GEN).mk:
 	     -f '{{range .Deps}}{{.}}{{"\n"}}{{end}}'          \
 	     ./cmd/libs/go2idl/conversion-gen                  \
 	     | grep "^$(PRJ_SRC_PATH)"                         \
+	     | sed 's|^$(PRJ_SRC_PATH)|./|'                    \
 	     | xargs go list -e -f '{{.Dir}} {{.Dir}}/*.go');  \
 	echo $$DIRECT $$INDIRECT | sed 's/ / =,/g' | tr '=,' '\n\t'; \
 	) | sed "s|$$(pwd -P)/||" > $@.tmp;                    \

--- a/hack/run-in-gopath.sh
+++ b/hack/run-in-gopath.sh
@@ -1,0 +1,33 @@
+#!/bin/bash
+
+# Copyright 2016 The Kubernetes Authors.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+# This script sets up a temporary Kubernetes GOPATH and runs an arbitrary
+# command under it.  Go tooling requires that the current directory be under
+# GOPATH or else it fails to find some things, such as the vendor directory for
+# the project.
+
+set -o errexit
+set -o nounset
+set -o pipefail
+
+KUBE_ROOT=$(dirname "${BASH_SOURCE}")/..
+source "${KUBE_ROOT}/hack/lib/init.sh"
+
+# This sets up a clean GOPATH and makes sure we are currently in it.
+kube::golang::setup_env
+
+# Run the user-provided command.
+"${@}"

